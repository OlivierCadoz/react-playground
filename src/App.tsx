import { BrowserRouter } from 'react-router';

import Header from '@components/molecules/Header';
import Routes from '@routes/index';

function App() {
  return (
    <>
      <BrowserRouter>
        <Header />

        <main>
          <Routes />
        </main>
      </BrowserRouter>
    </>
  );
}

export default App;
