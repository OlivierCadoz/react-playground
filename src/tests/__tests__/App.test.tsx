import App from '@/App';
import { render } from '@testing-library/react';

describe('App', () => {
  describe('When App is rendered', () => {
    test('Then it should match snapshot', () => {
      const { container } = render(<App />);

      expect(container).toMatchSnapshot();
    });
  });
});
