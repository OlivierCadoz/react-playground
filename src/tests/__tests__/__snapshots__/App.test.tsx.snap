// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > When App is rendered > Then it should match snapshot 1`] = `
<div>
  <header>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            data-discover="true"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=""
            data-discover="true"
            href="/morpion"
          >
            Morpion
          </a>
        </li>
        <li>
          <a
            class=""
            data-discover="true"
            href="/sudoku"
          >
            Sudoku
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <div>
      Home
    </div>
  </main>
</div>
`;
